src/WorkOS.net/Services/Webhooks/WebhookService.cs

```diff
@@ -0,0 +1,219 @@
+namespace WorkOS
+{
+    using System;
+    using System.Security.Cryptography;
+    using System.Text;
+    using Newtonsoft.Json;
+
+    public class WebhookService
+    {
+        private const int DefaultTimeTolerance = 300;
+
+        public static bool ConstantTimeAreEqual(byte[] a, byte[] b)
+        {
+            int i = a.Length;
+            if (i != b.Length)
+            {
+                return false;
+            }
+
+            int cmp = 0;
+            while (i != 0)
+            {
+                --i;
+                cmp |= a[i] ^ b[i];
+            }
+
+            return cmp == 0;
+        }
+
+        /// <summary>
+        /// Parses a JSON string from a webhook, verifies and deseralizes into a Webhook Object.
+        /// </summary>
+        /// <param name="json">The JSON string to parse.</param>
+        /// <param name="signatureHeader">
+        /// The value of the header from the webhook request.
+        /// </param>
+        /// <param name="secret">The webhook endpoint's signing secret.</param>
+        /// <param name="tolerance">The time tolerance, in seconds.</param>
+        /// <returns>The deserialized JSON.</returns>
+        public Webhook ConstructEvent(
+            string json,
+            string signatureHeader,
+            string secret,
+            long tolerance)
+        {
+            if (this.VerifyHeader(json, signatureHeader, secret, tolerance))
```

Comment:
This conditional could be a bit cleaner with a guard clause:

```cs
if (!this.VerifyHeader(json, signatureHeader, secret, tolerance))
{
    throw new Exception("Unable to verify Webhook");
}

return JsonConvert.DeserializeObject<Webhook>(json);
```

Also, if we end up going the route of making `VerifyHeader` return `void` then we don't need the conditional at all:

```cs
this.VerifyHeader(json, signatureHeader, secret, tolerance);

return JsonConvert.DeserializeObject<Webhook>(json);
```

Reply:
Went route #2, changed verifyheader to void and got rid of conditionals